### DISCUSSION TITLE
Template & Client Compatibility

### DISCUSSION THREAD

**Discussion by dm17** [voidlabs/mosaico#663] [Q&A]
I'm trying to find information on what makes Mosaico or MailChimp "more compatible with more email clients" - as I've seen many claims on this topic. 

1) How do we know that the many email clients in the wild will display our mosaico/versafix-composed email as accurately as they display MailChimp-composed emails? 

2) How can such a thing be known besides the amount of testing MailChimp must have done, considering its popularity?

3) What makes a template compatible with an SMTP relay? For example, if we compose with Mosaico and then send with Sparkpost or Sendgrid...

4) Are MailChimp and Mosaico templates cross compatible?

5) Considering the reporting features on Mosaico: how do the developers consider optimal integration with an SMTP relay, which does its own analytics and reporting?

Apologies, but this area is a bit opaque to me - and it seems this is true for many because of the lack of details in articles about newsletter compositing, WYSIWYG editors, and "compatible templates."

**Comment by bago** [comment]
A) Mosaico is template based, so compatibility depends on the template being used.
B) It is really unfair to think that Mailchimp compatibility is better just because they are big: just look in the internet to see WHO publish reports about email client issues and you won't see Mailchimp, there, but minor developers. E.g. the infamous 1px pixel bug from Outlook have been first reported by ME, even if it appeared for every sender, including Mailchimp ( https://mosaico.io/email-client-tricks/outlook-2016-weird-1px-horizontal-lines/ ). The most compatible technique to date, aka Fab4 ( https://mosaico.io/email-client-tricks/fab4-responsive-beyond-gmail/ ), have been shown by Rémi Parmentier aka HTeuMeuLeu, the author of "caniemail" ( https://www.caniemail.com/ ): no big email marketing service names like Mailchimp behind those stuff, because real innovation/evolution happens elsewhere.
C) I'd bet versafix-1, our master template, to show better compatibility than anything you will find around: of course everyone is good at most common clients, but you should check edge use cases, like gmail webmail from a smartphone (not the gmail app, but the gmail webapp through the mobile browser), or gmail app reading a non gmail email via IMAP (GANGA).

To answer your question:
1) I'd bet mosaico's "versafix-1" is more accurate (more compatible): do your own testing and report issues to the versafix-1 repo, if you find mailchimp is better: the more edge cases (e.g: country specific webmails/app) you'll test the more it will be clear.
2) wrong assumption.
3) this is a matter of what smtp relay service do: they should not break html, but if they do it is an issue in their service: we don't care of them: it is already a PITA to care of most email clients. If your smtp relay service changes your html when sending, simply switch to another service.
4) I don't know what you mean by this question.
5) "reporting features"? Which reporting features? Mosaico is just an email composer.

**Reply by dm17** [reply]
Thanks for moving to discussion & very useful!

A) So is there a Mosaico smoketest for templates, so we can get a reasonable idea what it might break before sending it to thousands of email addresses?

1&2) Makes sense

3) Yes but all of the SMTP relays have opaque marketing. Would be great if there was some FOSS groups (like this), which certified STMP relay services. The confusion comes from pages like this: https://www.sparkpost.com/blog/mandrill-template-migration-tool-mst3k/
It makes no sense why that would be needed unless they weren't actually compatible. I asked SparkPost and they wouldn't give me any meaningful reply. I agree of course: SMTP relays should be totally agnostic to the above discussion.

4) I guess there isn't as much of a need for template compatibility if we go with Mosaico because the default templates let you quickly build a variety of email types. This question was for people who find some great MailChimp templates and want to import them.

5) My bad - I misread. The email campaign reporting is within the SMTP relay and no need to have it within Mosaico.

Any idea if we could launch Mosaico in an iframe? I'm looking into how to integrate it with our webapp. The main functions we need over the default is to select which of our mailing lists to send them to - and build a gate so the email must sent to a test list before a production list. Hopefully that makes sense...

(Speaking of editors... Sorry I couldn't get both the indentation and numbering correct above, not sure on GitHubs auto formatting)

**Reply by bago** [reply]
Again, this is not a mosaico issue, but depending only on the master-template. We test versafix-1 master template against dozens (or maybe hundreds) of email clients/configurations. If you want to report compatibility issues for versafix-1 you can go here: https://github.com/voidlabs/versafix-template/issues

You won't find smoketests because every email is different and email clients are thousands: do identify your typical target and do your own testing, maybe by using litmus/emailonacid/testi@ or a similar service, if you want to save some time with a bunch of email clients.

3) this is out of scope for mosaico. We already spend time creating the email builder, we'll leave to anyone else to test smtp relay services. We never used an smtp relay, we don't know that world. If they break the HTML they send they are broken, IMHO.

4) there is no "import template" in mosaico, so I don't understand your question. You can create your own templates to be used in mosaico and if you do that the compatibility will be dependant on your own template and not to mosaico.

Yes, we expect you to run mosaico either full page or in an almost full page frame.

**Reply by bago** [reply]
Just to be clear, mosaico is free open source software, mailchimp is a paid service: you should better ask Mailchimp to prove how they do better with client compatibility ;-) Let us know their answer.

**Reply by dm17** [reply]
3) Yes, it is out of scope - and I'm not asking to increase scope. Just trying to comprehend how people use this - and what experts (like you) think the ideal configuration is before spending so much time & efforting going down a path. So, how do you send emails to thousands of people? I'd like to hear how Mosaico users who have 10K+ mailing lists do it. SMTP Relay seems - either self-hosted or not - seems like the only way to do it.

Not asking for any Mosaico improvements... Just a discussion to help people who are new to this area as there are some confusing aspects.  Perhaps I'm the only one confused about huge SMTP relay services blogging about template converters...

**Reply by bago** [reply]
Usually people already have a way to send emails and just need a modern email composer.

We (mosaico authors) are an italian ESP (https://www.voxmail.it/) and we have our own email servers.
CiviCRM (https://civicrm.org/) let their users to use mosaico, and I guess they use internal services by CiviCRM to send email via SMTP.
Mailtrain (https://github.com/Mailtrain-org/mailtrain) embeds mosaico and then use their own smtp services;
Lepatron (https://www.lepatron.email/) uses mosaico and exports the results to third party services (they also publish a github repo for ESP issues with html: https://github.com/Badsender-com/esp-html-bugs ) ;
Being free&opensource we don't know most of our users.

Sending massive emails is not something we care in mosaico: there are plenty of services or opensource tools to send emails. We care about building emails: it is a very specific goal. We don't even provide the backend needed by mosaico to deal with image upload/processing because of this.

**Reply by dm17** [reply]
Looks like I'll be exploring if Mosaico can be run in an iframe for our purposes. If not, then we'll just open a new window for it.
If possible, then I'd also like to get your comments on the email composing aspects of Ghost as compared to Mosaico:
https://ghost.org/help/email-newsletters/

**Reply by bago** [reply]
I can't answer, I never tried ghost email composer, but AGAIN, what makes Mosaico unique is it being "template based", so you can probably create a template for mosaico that let you create the same output you get from ghost email composer, but you can't customize how ghost email composer creates its own email.

From the page you linked seems like their composer is very light and let you create very simple single column newsletters.

**Reply by dm17** [reply]
Thanks - this gives me a better feel for where this can fit in the overall system. The standard way to embed videos with Mosaico is to link to the image (video thumbnail) and self-host it? I was hoping there's a more elegant way of doing this for our next emailing flow.

**Reply by bago** [reply]
Mosaico doesn't care/deal with this. versafix-1 template doesn't give you a "video block". You can easily create a template that will embed hosted videos, if you know how to do that and you are happy with email client support of your solution.

Most email clients do not support video in email so most senders simply send an image and link that image to an online video (youtube, vimoe or whatever you like). Some email builder automagically create an "image block" starting from a youtube video url (extract an image thumbnail of the video and simply put an image with a link to the video): mosaico does not provide this facility.

https://www.caniemail.com/features/html-video/
